resteasy-client/src/main/java/org/jboss/resteasy/client/jaxrs/internal/proxy/ClientInvoker.java

```diff
@@ -77,11 +82,87 @@ public Method getResourceMethod()
          feature.configure(info, new FeatureContextDelegate(invokerConfig));
       }
 
-
-      this.processors = ProcessorFactory.createProcessors(declaring, method, invokerConfig, config.getDefaultConsumes());
-      accepts = MediaTypeHelper.getProduces(declaring, method, config.getDefaultProduces());
       entityExtractorFactory = new DefaultEntityExtractorFactory();
       this.extractor = entityExtractorFactory.createExtractor(method);
+
+      MediaType defaultConsumes = checkConsumesMediaTypes(config);
+
+      this.processors = ProcessorFactory.createProcessors(declaring, method, invokerConfig, defaultConsumes);
+      accepts = MediaTypeHelper.getProduces(declaring, method, config.getDefaultProduces());
+
+      checkProducesMediaTypes();
+   }
+
+   /**
+    * Traverse the parent tree checking for a class associated Consumes annotation.
+    * @param config
+    * @return
+    */
+   private MediaType checkConsumesMediaTypes(ProxyConfig config)
+   {
+
+      MediaType defaultConsumes = config.getDefaultConsumes();
+      if (defaultConsumes == null)
+      {
+         WeightedMediaType weightedMediaType = null;
+         Queue<Class<?>> interfaces = new LinkedList<>();
+         for (Class<?> superClass = declaring; superClass != null; superClass = superClass.getSuperclass())
+         {
+            Collections.addAll(interfaces, superClass.getInterfaces());
+            for (Class<?> next = interfaces.poll(); next != null; next = interfaces.poll())
+            {
+               Consumes annotation = (Consumes) next.getAnnotation(Consumes.class);
+               if (annotation != null) {
+                  String[] values = annotation.value();
+                  for (String value : values)
+                  {
+                     WeightedMediaType candidateWeightedMediaType = WeightedMediaType.valueOf(value);
+                     if (weightedMediaType == null || candidateWeightedMediaType.compareTo(weightedMediaType) < 0) {
+                        weightedMediaType = candidateWeightedMediaType;

```

Comment:
Can you explain me the reasoning behind this please?
And isn't the new checkConsumesMediaTypes(..) method here changing the behavior in case of defaultConsumes being null?

Reply:
When there is no @Consumes annotation on the current class, resteasy should
check the super-class for the @Consumes annotation.  Because @Consumes has 
annotation @Inherited, every sub-class should inherit the super-class's @Consumes 
annotation when the sub-class does not have a @Consumes annotation declared.
However, when the current sub-class has a @Consumes annotation it overrides 
the parent annotation.